klippy/extras/output_pin.py

```diff
@@ -45,18 +46,23 @@ def get_status(self, eventtime):
     def cmd_SET_PIN(self, gcmd):
         value = gcmd.get_float('VALUE', minval=0., maxval=self.scale)
         value /= self.scale
-        if value == self.last_value:
-            return
         print_time = self.printer.lookup_object('toolhead').get_last_move_time()
-        print_time = max(print_time, self.last_value_time + PIN_MIN_TIME)
+        print_time = max(print_time, self.last_print_time + PIN_MIN_TIME)
         if self.is_pwm:
-            self.mcu_pin.set_pwm(print_time, value)
+            cycle_time = gcmd.get_float('CYCLE_TIME',
+                self.mcu_pin._cycle_time, above=0.)
```

Comment:
It's not valid to peek into another classes member variables.  (As mentioned in docs/Code_Overview.md, doing so creates a maintenance burden.)  Either store cycle_time locally, or add a get_cycle_time() method to the MCU_pwm class.

Reply:
I'll store it locally as default_cycle_time. I wasn't aware of the convention that names starting with an underscore are considered private.